File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
class Solution:
    def invertTree(self, root: Optional[Treeroot]) -> Optional[Treeroot]:
        if(not root):
            return None
        
        root.left, root.right = root.right, root.left
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.cpp
class Solution {
private:
    bool sameTree(TreeNode* s, TreeNode* t){
        if(!s && !t){ return true; }

        if(s && t && s->val == t->val){
            return sameTree(s->left, t->left) && sameTree(s->right, t->right);
        }
        return false;
    }
public:
    bool isSubtree(TreeNode* root, TreeNode* subRoot) {
        if(!subRoot){ return true; }
        if(!root){ return false; }

        if (sameTree(root, subRoot)){ return true; }

        return (isSubtree(root->left, subRoot) || isSubtree(root->right, subRoot));
    }
};
